docs/advanced-examples.md

```diff
@@ -1 +1,344 @@
+}
+```
+
+As such, you will need to engineer your pipeline in a way that the plugins transform your input format into this final format. This document has a list of commonly used plugins and example scenarios in which they can be used.
+
+If you need a way to do something not listed here, we've made it easy to create custom plugins! See our documentation on [creating a new pipeline plugin in Java/Kotlin](/docs/writing-java-kotlin-plugin.md) or [creating a new pipeline plugin in a different language](/docs/writing-custom-plugin.md).
+
+## Document structure
+
+This document has two sections. First, there is a list of plugins and example scenarios in which they may be used. These each describe the plugins workings on a simple level. The [plugin index](#plugin-index) has a full list of the plugins and the scenarios in which they may be used.
+
+Secondly, the (example pipelines)[#example-pipelines] section contains a set of example input payloads and the recommended set of plugins that can be used to transform them to the required output format. They are meant to be abridged versions of the [main Hyperion pipeline tutorial](/docs/hyperion-setup.md), and as such do not detail full configuration files but rather a list of plugins and references to their appropriate documentation.
+
+## Plugin Index
+
+I want to...
+
+- [Inspect the current messages going through the pipeline](#plugin-printer)
+- [Measure the rate of messages going through the pipeline](#plugin-rate)
+- [Rename or move a field](#plugin-renamer)
+- [Add a statically known value to the payload](#plugin-adder)
+- [Extract string values from a field into their own fields](#plugin-extractor)
+- [Transform a Java package name to a file path](#plugin-pathextractor)
+- [Attach a version tag based on the latest commit in a Git repository](#plugin-versiontracker)
+- [Load balance multiple instances of a single plugin to increase pipeline throughput](#plugin-loadbalancer)
+- [Read messages from stdin and send them through the pipeline to debug my plugin setup](#plugin-reader)
+- [Stress-test my pipeline by sending a huge amount of messages](#plugin-stresser)
+
+Want to do something not listed here? You might be able to do it by combining multiple plugins. If that doesn't work, check out our documentation on [creating a new pipeline plugin in Java/Kotlin](/docs/writing-java-kotlin-plugin.md) and [creating a new pipeline plugin in a different language](/docs/writing-custom-plugin.md).
+
+### Plugin: Printer
+
+The [printer](/pipeline/plugins/printer) plugin is a useful tool for anyone that wishes to debug the messages going through their pipeline. As the name suggests, it simply prints any incoming messages to stdout before forwarding it to the next plugin. Use it to get an idea of the type of data flowing through your pipeline.
+
+Due to its simplicity, the printer plugin has no additional configuration. If needed though, you can find the documentation [here](/pipeline/plugins/printer/README.md).
```

Comment:
It might be cleaner to just link to `/pipeline/plugins/printer` here since the README is automatically visible from that location? This is up for debate however.

Reply:
Perhaps match this sentence to all the others as well. As an example:
`For more information and the full set of configuration options, please see the [printer documentation](/pipeline/plugins/printer).`